modules/ml/src/main/java/org/apache/ignite/ml/tree/impurity/ImpurityMeasureCalculator.java

```diff
@@ -18,21 +18,71 @@
 package org.apache.ignite.ml.tree.impurity;
 
 import java.io.Serializable;
+import org.apache.ignite.ml.tree.TreeFilter;
 import org.apache.ignite.ml.tree.data.DecisionTreeData;
+import org.apache.ignite.ml.tree.data.TreeDataIndex;
 import org.apache.ignite.ml.tree.impurity.util.StepFunction;
 
 /**
  * Base interface for impurity measure calculators that calculates all impurity measures required to find a best split.
  *
  * @param <T> Type of impurity measure.
  */
-public interface ImpurityMeasureCalculator<T extends ImpurityMeasure<T>> extends Serializable {
+public abstract class ImpurityMeasureCalculator<T extends ImpurityMeasure<T>> implements Serializable {
```

Comment:
Do you think it's a good idea to remove high-level abstraction like `ImpurityMeasureCalculator` and replace it with something that has index? What about calculators without index?

It's not so important, but from design perspective looks not good.

Reply:
leave it as is